### DISCUSSION TITLE
Help with MILP model

### DISCUSSION THREAD

**Discussion by ConorRobinson78** [dirkschumacher/ompr#332] [Q&A]
Hi Dirk,

I am attempting to solve a MILP problem with multiple objectives. One is to minimize the transportation cost based on a matrix of customers to plants (caj) and then adding this to the objective function which is to maximize profit given a set of bid amounts from customers, fixed costs of 70000 per plant and unit cost of 10 per order quantities.

I have been trying to solve this for some time now and keep getting an error message
"Error in ast[[i]] : subscript out of bounds"

I cannot seem to derive a solution due to this error. Any help would be much appreciated

[rscrip-ontgifter.txt](https://github.com/dirkschumacher/ompr/files/6157063/rscrip-ontgifter.txt)

**Comment by sbmack** [comment]
Firstly, your objective function is messed up:

```
P1 <- MIPModel() %>%
  
  add_variable(x[a,j], a = 1:8, j= 1:8, type = "binary") %>% #  if Plant (a) gets assigned to customer (j)
  add_variable(y[a], type = "integer", a = 1:8) %>% #if plant is used
  
  set_objective(sum_expr(colwise(bid[j], j=1:8))) -   # no variable x or y in this term, need to wrap colwise in parens: colwise(bid[j])
  sum_expr(transportcost(a,j) * x[a,j], a=1:8, j=1:8) - 
  sum_expr(colwise(ord[j]) *10, j = 1:8) %>%  # no variable x or y in this term
```
Also `colwise` is a function required for some MILPModel formulating, (MIPModel here).  Don't know how that may effect the processing of your MIPModel when the missing variables are added.
